crates/isograph_schema/src/validate_server_field.rs

```diff
@@ -62,11 +76,20 @@ fn validate_and_transform_server_field(
                 description: empty_field.description,
                 name: empty_field.name,
                 id: empty_field.id,
-                associated_data: field_type,
+                associated_data: match field_type.inner_non_null() {
+                    SelectionType::Scalar(scalar_id) => {
+                        SelectionType::Scalar(field_type.map(&mut |_| scalar_id))
```

Comment:
Can we panic if the variant is an `InlineFragment` here? (Maybe that will eventually be an error, but for now it's a bug so we should panic)